core/iwasm/interpreter/wasm_interp_fast.c

```diff
@@ -5646,27 +5650,151 @@ wasm_interp_call_func_bytecode(WASMModuleInstance *module,
 #endif
                 goto call_func_from_entry;
             }
-#if WASM_ENABLE_SIMD != 0
+#if WASM_ENABLE_SIMDE != 0
+#define SIMD_V128_TO_SIMDE_V128(v)                                      \
+    ({                                                                  \
+        bh_assert(sizeof(V128) == sizeof(simde_v128_t));                \
+        simde_v128_t result;                                            \
+        bh_memcpy_s(&result, sizeof(simde_v128_t), &(v), sizeof(V128)); \
+        result;                                                         \
+    })
+
+#define SIMDE_V128_TO_SIMD_V128(sv, v)                                \
+    do {                                                              \
+        bh_assert(sizeof(V128) == sizeof(simde_v128_t));              \
+        bh_memcpy_s(&(v), sizeof(V128), &(sv), sizeof(simde_v128_t)); \
+    } while (0)
+
             HANDLE_OP(WASM_OP_SIMD_PREFIX)
             {
                 GET_OPCODE();
 
                 switch (opcode) {
                     /* Memory */
                     case SIMD_v128_load:
+                    {
```

Comment:
In my view, wouldn't it be better to move all the new ~1000 lines to a new, separate file? The original core/iwasm/interpreter/wasm_interp_fast.c is already large. let's not make it any bigger.

Reply:
I agree - I'm happy to follow up with a PR that moves things around later, but I have a few other branches that I'd need to rebase etc so would be good to merge this? This is just a feature branch so I think it's okay if it's a bit messy for now?

Reply:
@jammar1 I added several comments, could you have a look? I am fine to merge this PR to the branch `dev/simd_for_interp` first but we should address the issues then. Do you want to merge this PR first, or merge it after you address the new comments?

Reply:
@wenyongh I've addressed the comments and CI/ the checks have passed. Could you merge it if you're happy with the changes? Thanks